### PR TITLE
Head urdf fix for cob4-8

### PR DESCRIPTION
**Pull request by MattiaRacca**
- fixes head urdf configuration on the cob-aalto (cob4-8)

### CODE REVIEW

**Comment by ipa-rmb** [comment]
@fmessmer @floweisshardt Do you have any objections merging this? I do not have any.

File: cob_hardware_config/robots/cob4-8/urdf/calibration_default.urdf.xacro [hunk 1]
```diff
@@ -46,14 +46,25 @@
 	<xacro:property name="def_head_roll" value="0.0"/>
 	<xacro:property name="def_head_pitch" value="0.0"/>
 	<xacro:property name="def_head_yaw" value="0.0"/>
-
+    
+    <!-- Legacy transformation, following from WRONG urdf-->
```
Comment (ipa-rmb) [inline cob_hardware_config/robots/cob4-8/urdf/calibration_default.urdf.xacro]: Please correct indentation here before we can merge.

**Review by ipa-rmb — DISMISSED** [review]

File: cob_hardware_config/robots/cob4-8/urdf/cob4-8.urdf.xacro [hunk 2]
```diff
@@ -71,7 +71,7 @@
     <origin xyz="${head_cam_x} ${head_cam_y} ${head_cam_z}" rpy="${head_cam_roll} ${head_cam_pitch} ${head_cam_yaw}" />
   </xacro:usb_cam>
 
-  <xacro:sensorring name="sensorring" parent="head_3_link" active="true">
+  <xacro:sensorring name="sensorring" parent="head_2_link" active="true">
```
Comment (fmessmer) [inline cob_hardware_config/robots/cob4-8/urdf/cob4-8.urdf.xacro:74]: if this is fixing the kinematic description of the head- sensorring-combo, we might need to apply this to all robots with such combo

**Review by fmessmer — COMMENTED** [review]
thanks @MattiaRacca for the PR

I'll have to review the tf transforms before and with this PR

also, have you used this description also in simulation (gazebo)?
do the tf behave the same when executing the same motion in SIM vs. HW?
could you verify this by e.g. recording `/joint_states` and `/tf` when executing the same knoeppkes sequence?

File: cob_hardware_config/robots/cob4-8/urdf/cob4-8.urdf.xacro (hunk 2 above)
Comment (MattiaRacca) [reply cob_hardware_config/robots/cob4-8/urdf/cob4-8.urdf.xacro:74]: Hi @fmessmer,

we checked that the motions are the same in HW and in Rviz and they overlap (no gazebo though). I unfortunately can't verify that now as I am in the US for the next 4 months (I fixed this "head issue" before leaving). I will talk to my colleagues in Helsinki and see if somebody can do this check for you.

Comment (MattiaRacca) [reply cob_hardware_config/robots/cob4-8/urdf/cob4-8.urdf.xacro:74]: Hi @fmessmer,
My colleague Jens Lundell promised to take over from here. I'll write you an email to put you two in contact.

**Comment by fmessmer** [comment]
there is more messed up on the head urdf...see https://github.com/ipa320/cob_common/pull/284

as we currently have the robot for repair anyway, we'll finalize the issue in #795

**Pull request closed without merging**